README.md

```diff
@@ -24,23 +24,38 @@ Cert manager is needed to provision KFServing webhook certs for production grade
 generation [script](./hack/self-signed-ca.sh).
 
 ### Install KFServing
+#### Standalone KFServing Installation
+KFServing can be installed standalone if your kubernetes cluster meets the above prerequisites and KFServing controller is deployed in `kfserving-system` namespace.
 ```
 TAG=0.2.2
 kubectl apply -f ./install/$TAG/kfserving.yaml
 ```
-By default, you can create InferenceService instances in any namespace which has no label with `control-plane` as key.
-You can also configure KFServing to make InferenceService instances only work in the namespace which has label pair `serving.kubeflow.org/inferenceservice: enabled`. To enable this mode, you need to add `env` field as stated below to `manager` container of statefulset `kfserving-controller-manager`.
+KFServing uses pod mutator or [mutating admission webhooks](https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/) to inject the storage initializer component of KFServing. By default all the pods in namespaces which are not labelled with `control-plane` label go through the pod mutator.
+This can cause problems and interfere with Kubernetes control panel when KFServing pod mutator webhook is not in ready state yet.
 
+For Kubernetes 1.14 users we suggest enabling the following environment variable `ENABLE_WEBHOOK_NAMESPACE_SELECTOR` so that only pods
+ in the namespaces which are labelled `serving.kubeflow.org/inferenceservice: enabled` go through the KFServing pod mutator.
 ```
 env:
 - name: ENABLE_WEBHOOK_NAMESPACE_SELECTOR
   value: enabled
 ```
-### Install KFServing in 5 Minutes (On your local machine)
+
+For Kubernetes 1.15+ users we strongly suggest turning on the [object selector](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#matching-requests-objectselector) so that only KFServing `InferenceService` pods go through the pod mutator.
+```bash
+kubectl patch mutatingwebhookconfiguration inferenceservice.serving.kubeflow.org --patch '{"webhooks":[{"name": "inferenceservice.kfserving-webhook-server.pod-mutator","objectSelector":{"matchExpressions":[{"key":"serving.kubeflow.org/inferenceservice", "operator": "Exists"}]}}]}'
+```
+#### KFServing in Kubeflow Installation
+KFServing is installed by default as part of Kubeflow installation using [Kubeflow manifests](https://github.com/kubeflow/manifests/tree/master/kfserving) and KFServing controller is deployed in `kubeflow` namespace.
+Since Kubeflow kubernetes minimal requirement is 1.14 which does not support object selector, `ENABLE_WEBHOOK_NAMESPACE_SELECTOR` is enabled in Kubeflow installation by default
```

Comment:
```suggestion
Since Kubeflow Kubernetes minimal requirement is 1.14 which does not support object selector, `ENABLE_WEBHOOK_NAMESPACE_SELECTOR` is enabled in Kubeflow installation by default.
```